from PyQt5 import QtCore, QtGui, QtWidgets
import sqlite3
from loantype import *
from declaration2 import *
from datetime import date
import time

class Ui_loan_request(object):
    def setupUi(self, loan_request):
        loan_request.setObjectName("loan_request")
        loan_request.resize(1368, 878)
        loan_request.setMinimumSize(QtCore.QSize(1368, 878))
        loan_request.setMaximumSize(QtCore.QSize(1368, 878))
        self.centralwidget = QtWidgets.QWidget(loan_request)
        self.centralwidget.setObjectName("centralwidget")
        self.F1 = QtWidgets.QFrame(self.centralwidget)
        self.F1.setGeometry(QtCore.QRect(-20, -10, 1371, 881))
        self.F1.setStyleSheet("background-color:rgb(255, 170, 255);")
        self.F1.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.F1.setFrameShadow(QtWidgets.QFrame.Raised)
        self.F1.setObjectName("F1")
        self.F2 = QtWidgets.QFrame(self.F1)
        self.F2.setGeometry(QtCore.QRect(10, 10, 251, 861))
        self.F2.setStyleSheet("background-color: rgb(64, 54, 200);\n"
"border:20px;\n"
"border-radius:10px;")
        self.F2.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.F2.setFrameShadow(QtWidgets.QFrame.Raised)
        self.F2.setObjectName("F2")
        self.label = QtWidgets.QLabel(self.F1)
        self.label.setGeometry(QtCore.QRect(510, 40, 161, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.F1)
        self.label_2.setGeometry(QtCore.QRect(530, 100, 91, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.acnotext = QtWidgets.QLineEdit(self.F1)
        self.acnotext.setGeometry(QtCore.QRect(680, 90, 261, 41))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.acnotext.setFont(font)
        self.acnotext.setStyleSheet("background-color: rgb(255, 255, 255);\n"
"")
        self.acnotext.setText("")
        self.acnotext.setObjectName("acnotext")
        self.banknamecombo = QtWidgets.QComboBox(self.F1)
        self.banknamecombo.setGeometry(QtCore.QRect(680, 30, 261, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.banknamecombo.setFont(font)
        self.banknamecombo.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.banknamecombo.setObjectName("banknamecombo")
        self.label_3 = QtWidgets.QLabel(self.F1)
        self.label_3.setGeometry(QtCore.QRect(530, 160, 111, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.ifsctext = QtWidgets.QLineEdit(self.F1)
        self.ifsctext.setGeometry(QtCore.QRect(680, 150, 261, 41))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.ifsctext.setFont(font)
        self.ifsctext.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.ifsctext.setText("")
        self.ifsctext.setObjectName("ifsctext")
        self.searchbut = QtWidgets.QPushButton(self.F1)
        self.searchbut.setGeometry(QtCore.QRect(700, 220, 171, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.searchbut.setFont(font)
        self.searchbut.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.searchbut.setStyleSheet("background-color: rgb(194, 194, 194);")
        self.searchbut.setObjectName("searchbut")
        self.table = QtWidgets.QTableWidget(self.F1)
        self.table.setGeometry(QtCore.QRect(380, 300, 741, 231))
        self.table.setStyleSheet("background-color: rgb(172, 231, 255);")
        self.table.setObjectName("table")
        self.table.setColumnCount(6)
        self.table.setRowCount(0)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.table.setHorizontalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.table.setHorizontalHeaderItem(1, item)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.table.setHorizontalHeaderItem(2, item)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.table.setHorizontalHeaderItem(3, item)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.table.setHorizontalHeaderItem(4, item)
        item = QtWidgets.QTableWidgetItem()
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        item.setFont(font)
        self.table.setHorizontalHeaderItem(5, item)
        self.requestbut = QtWidgets.QPushButton(self.F1)
        self.requestbut.setGeometry(QtCore.QRect(590, 720, 161, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.requestbut.setFont(font)
        self.requestbut.setStyleSheet("background:\n"
"rgb(193, 193, 193)")
        self.requestbut.setObjectName("requestbut")
        self.label_4 = QtWidgets.QLabel(self.F1)
        self.label_4.setGeometry(QtCore.QRect(510, 630, 141, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.amounttext = QtWidgets.QLineEdit(self.F1)
        self.amounttext.setGeometry(QtCore.QRect(670, 630, 261, 41))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.amounttext.setFont(font)
        self.amounttext.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.amounttext.setObjectName("amounttext")
        self.label_5 = QtWidgets.QLabel(self.F1)
        self.label_5.setGeometry(QtCore.QRect(510, 560, 141, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.loantypecombo = QtWidgets.QComboBox(self.F1)
        self.loantypecombo.setGeometry(QtCore.QRect(670, 560, 261, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.loantypecombo.setFont(font)
        self.loantypecombo.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.loantypecombo.setObjectName("loantypecombo")
        self.otherbut = QtWidgets.QPushButton(self.F1)
        self.otherbut.setGeometry(QtCore.QRect(940, 570, 111, 31))
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.otherbut.setFont(font)
        self.otherbut.setStyleSheet("background-color: rgb(196, 188, 198);")
        self.otherbut.setObjectName("otherbut")
        self.dec = QtWidgets.QPushButton(self.F1)
        self.dec.setGeometry(QtCore.QRect(850, 720, 161, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.dec.setFont(font)
        self.dec.setStyleSheet("background-color: rgb(200, 200, 200);")
        self.dec.setObjectName("dec")
        loan_request.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(loan_request)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1368, 27))
        self.menubar.setObjectName("menubar")
        loan_request.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(loan_request)
        self.statusbar.setObjectName("statusbar")
        loan_request.setStatusBar(self.statusbar)

        self.retranslateUi(loan_request)
        QtCore.QMetaObject.connectSlotsByName(loan_request)
        
        self.searchbut.clicked.connect(self.search)
        self.otherbut.clicked.connect(self.link)
        self.dec.clicked.connect(self.declaration)
        self.showbn()
        self.showlt()
        self.dec.setVisible(True)
        self.requestbut.setVisible(False)
        self.requestbut.clicked.connect(self.reqBTN)

        self.retranslateUi(loan_request)
        QtCore.QMetaObject.connectSlotsByName(loan_request)

    def retranslateUi(self, loan_request):
        _translate = QtCore.QCoreApplication.translate
        loan_request.setWindowTitle(_translate("loan_request", "Loan_Request"))
        self.label.setText(_translate("loan_request", "BANK NAME:"))
        self.label_2.setText(_translate("loan_request", "ACC.NO:"))
        self.label_3.setText(_translate("loan_request", "IFSC CODE:"))
        self.searchbut.setText(_translate("loan_request", "SEARCH"))
        item = self.table.horizontalHeaderItem(0)
        item.setText(_translate("loan_request", "NAME"))
        item = self.table.horizontalHeaderItem(1)
        item.setText(_translate("loan_request", "EMAIL"))
        item = self.table.horizontalHeaderItem(2)
        item.setText(_translate("loan_request", "PH.NO"))
        item = self.table.horizontalHeaderItem(3)
        item.setText(_translate("loan_request", "ADDRESS"))
        item = self.table.horizontalHeaderItem(4)
        item.setText(_translate("loan_request", "AADHAR.NO"))
        item = self.table.horizontalHeaderItem(5)
        item.setText(_translate("loan_request", "PAN.NO"))
        self.requestbut.setText(_translate("loan_request", "REQUEST"))
        self.label_4.setText(_translate("loan_request", "LOAN AMOUNT:"))
        self.label_5.setText(_translate("loan_request", "LOAN TYPE:"))
        self.otherbut.setText(_translate("loan_request", "Others"))
        self.dec.setText(_translate("loan_request", "DECLARATION"))

    def search(self):
        bn=self.banknamecombo.currentText()
        ac=self.acnotext.text()
        ifsc=self.ifsctext.text()
        con = sqlite3.connect("UMRS.db")
        cur = con.cursor()
        q = "select Name,Email,PhoneNo,Address,AadharNo,PanNo from CoustomerReg where BankName=? and ACNo=? and IFSCcode=?"
        cur.execute(q,(bn,ac,ifsc))
        res = cur.fetchall()
        if ac=="" or ifsc== "" :
            if ac=="":

                msg = QMessageBox()
                msg.setIcon(QMessageBox.Critical)
                msg.setText("A/C Number field cann't be empty")
                msg.setWindowTitle("Warning")
                msg.show()
                msg.exec_()
            elif ifsc=="":
                msg = QMessageBox()
                msg.setIcon(QMessageBox.Critical)
                msg.setText("IFSC code field cann't be empty")
                msg.setWindowTitle("Warning")
                msg.show()
                msg.exec_()

        else:
        # print(res)
            self.table.setRowCount(0)
            for i in res:
                rc = self.table.rowCount()
                self.table.insertRow(rc)
                self.table.setItem(rc,0,QtWidgets.QTableWidgetItem(str(i[0])))
                self.table.setItem(rc,1,QtWidgets.QTableWidgetItem(i[1]))
                self.table.setItem(rc,2,QtWidgets.QTableWidgetItem(str(i[2])))
                self.table.setItem(rc,3,QtWidgets.QTableWidgetItem(str(i[3])))
                self.table.setItem(rc,4,QtWidgets.QTableWidgetItem(i[4]))
                self.table.setItem(rc,5,QtWidgets.QTableWidgetItem(str(i[5])))
        con.commit()
            
    def link(self):
        # self.MainWindow = QtWidgets.QMainWindow()
        # self.ui = Ui_MainWindow()
        # self.ui.setupUi(self.MainWindow)
        # self.MainWindow.show()
        self.LoanType = QtWidgets.QMainWindow()
        self.ui = Ui_LoanType()
        self.ui.setupUi(self.LoanType)
        self.LoanType.show()

    def declaration(self):
        # bn = self.banknamecombo.currentText()
        # con = sqlite3.connect("UMRS.db")
        # cur = con.cursor()
        self.dec = QtWidgets.QMainWindow()
        self.ui = Ui_dec()
        self.ui.setupUi(self.dec)
        self.dec.show()

        # self.dec.setVisible(False)
        self.requestbut.setVisible(True)


    def showbn(self):
        bn = self.banknamecombo.currentText()
        con = sqlite3.connect("UMRS.db")
        cur = con.cursor()
        q1 = "select BankName from Bank_Name"
        cur.execute(q1,(bn))
        c = cur.fetchall()
        for i in c:
            self.banknamecombo.addItem(i[0])
        con.commit()

    def showlt(self):
        lt = self.loantypecombo.currentText()
        con = sqlite3.connect("UMRS.db")
        cur = con.cursor()
        q1 = "select LoanType from Loan_Type"
        cur.execute(q1,(lt))
        c = cur.fetchall()
        for i in c:
            self.loantypecombo.addItem(i[0])
        con.commit()

    def reqBTN(self):        
        bn=self.banknamecombo.currentText()
        ac=self.acnotext.text()
        ifsc=self.ifsctext.text()
        lt = self.loantypecombo.currentText()
        amt = self.amounttext.text()

        con = sqlite3.connect("UMRS.db")
        cur = con.cursor()
        q = "select Name,Email,PhoneNo,Address,AadharNo,PanNo from CoustomerReg where BankName=? and ACNo=? and IFSCcode=?"
        cur.execute(q,(bn,ac,ifsc))
        res = cur.fetchall()
        newList = []
        for i in res:
            for element in i:
                newList.append(element)

        # print(newList)
        name = newList[0]
        email = newList[1]
        phone = newList[2]
        address = newList[3]
        addhar = newList[4]
        pan = newList[5]

        dt = date.today()
        ti = time.strftime("%H:%M:%S",time.localtime())

        cur1 = con.cursor()
        q1 =  "Select LID from Loan_Request order by LID desc limit 1"
        cur1.execute(q1)
        res = cur1.fetchall()
        lid = res[0][0] + 1

        cur2 = con.cursor()
        q2 = "insert into Loan_Request values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
        cur2.execute(q2,(lid,name, email, phone, address,addhar,pan,bn,ac, ifsc,amt,lt,dt,ti))
        con.commit()
        msg = QMessageBox()
        msg.setIcon(QMessageBox.Information)
        msg.setText("Loan request successful")
        msg.setWindowTitle("Loan Request")
        msg.show()
        msg.exec_()


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    loan_request = QtWidgets.QMainWindow()
    ui = Ui_loan_request()
    ui.setupUi(loan_request)
    loan_request.show()
    sys.exit(app.exec_())
